### ISSUE TITLE
Multi-GPU DDP: Low batch size limit, slow validation and unbalanced VRAM usage on 8×A800 for RF-DETR-2XLarge

### ISSUE THREAD

**Issue by gzhex** [roboflow/rf-detr#1095]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

I'm really impressed by the accuracy of the newly released RF-DETR-2XLarge, so I immediately conducted experiments on my custom dataset. However, I encountered several issues when running distributed training on **8×A800 (80GB)** GPUs:
With the default input resolution of 880, the maximum **batch size I can set is only 4**. Any larger batch size will trigger an OOM error. This behavior is abnormal, as 8 A800 GPUs provide a total of 640GB VRAM, which should not be constrained to such a small batch size.
Even with batch size set to 4, the training time is unreasonable. The training process only takes 46 minutes, while validation via the Validation DataLoader takes over 2 hours.
I also observed unbalanced GPU workloads across devices. Most cards occupy around 40GB VRAM, while one or two individual cards consume more than 70GB of VRAM.

For example:
Epoch 0: 100%|█████████████████████████████████████████████████████| 2780/2780 [46:06<00:00,  1.01it/s, train/lr=0.0001, train/lr_min=3.23e-6, train/lr_max=0.0001]
Validation DataLoader 0:  63%|█████████████████████████████████████████████████████████████▏                                   | 251/398 [1:23:01<48:37,  0.05it/s]

### Environment

RF-DETR: 1.7.1
RF-DETR-PLUS: 1.0.2
OS: Ubuntu 20.04
Python: 3.10.19
PyTorch: 2.7.0+cu118
CUDA: 12.2
cuDNN: 9.1.0
GPU: NVIDIA A800-SXM4-80GB * 8

### Minimal Reproducible Example

python -m torch.distributed.run --nproc_per_node 8 --master_port 60847 train.py

import os
DATASET_DIR = "RF-DETR/datasets"
OUTPUT_DIR = "RF-DETR/output"
WEIGHTS_PATH = "RF-DETR/weights/rf-detr-xxlarge.pth"

EPOCHS = 30
BATCH_SIZE = 4
LEARNING_RATE = 1e-4
NUM_CLASSES = 51  # 2 real classes + 1 background
NUM_WORKERS = 0

from rfdetr_plus import RFDETR2XLarge

model = RFDETR2XLarge(pretrain_weights=WEIGHTS_PATH)

model.train(
    dataset_dir=DATASET_DIR,
    epochs=EPOCHS,
    batch_size=BATCH_SIZE,
    grad_accum_steps=1,
    devices=8,
    strategy="ddp_find_unused_parameters_true",
    num_workers=8,
    progress_bar="tqdm", 
    pin_memory=False,
    persistent_workers=False,
    prefetch_factor=1,
    output_dir=OUTPUT_DIR,
)

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by isaacrob** [comment]
how large are your images? during val it upsamples the masks to image res and then does RLE so it can compute mAP. very large images therefore lead to OOM and slow val

**Comment by isaacrob** [comment]
(just to confirm - the OOM you're seeing is during val, yes? this seems to be a recurring problem with the same core cause, very high resolution training images. perhaps we should change default behavior so this doesn't happen)

**Comment by gzhex** [comment]
I have resized the image to 880*880, which is as same as RF-DETR-2XLarge, but the original size may be 1K or 2K.
Yes, it's so slow during val, which is three times during training. And, one of the VRAM is up to 70G, which means i only can set batchsize 4.

**Comment by isaacrob** [comment]
I am not sure how else we can measure mAP without the upsample. The RLE encoding is what's slow and the upsample is what costs memory. Any thoughts?

Worth noting that you can increase effective batch size via gradient accumulation with the existing flags. We use layer norm instead of batch norm so it should be safe.

**Comment by anatoly-ryabchenko** [comment]
@isaacrob Do we need exact COCO-style mask mAP *during* training, or only at the end?
The mask loss is already computed at the head's native resolution, so the model only ever optimizes mask quality at that resolution. My intuition tells me that upsampling to original size would inherently fit the eval set during training, but will not add any improvement on the test set when evaluated at original map sizes. But honestly, I'd also expect it to pick the same "best" checkpoint most of the time, and we still get the true full-res number from the final/test eval at the end. If you are interested, I can implement and check that on some dataset(s) - I have unused modal and colab credits. :)

@gzhex also, you might notice that `num_select` - the number of masks the model outputs per image - is configurable. Each of those is upsampled to the full image size (that's the memory spike) and then RLE-encoded for the metric (that's the slow part). If in your case you don't need to keep all those masks during eval, you can lower `num_select` (at model definition) to cut eval memory and time and allow larger batches.
